### ISSUE TITLE
gg causes capture group to be lost

### ISSUE THREAD

**Issue by samuel1** [mawww/kakoune#1720]
`/hello<ret>yggP` puts hello at the top of the page
`/(hello)<ret>ygg"1P` puts nothing at the top of the page
`/(hello)<ret>y5000h5000k"1P` puts hello at the top of the page

It seems `gg` causes the capture group to be lost, whereas `5000h5000k` doesn't

**Comment by mawww** [comment]
That is correct, `gg` **replaces** selections with a new one, and as captures are stored as part of the selection, we lose them. `h` or `k` **moves** selections around, so captures are preserved.

The semantics of `gg` could be changed to `delete all selections except the main one, and move it to start of buffer`. I am not sure if this is desirable or not.

**Comment by samuel1** [comment]
Okay.  Thanks

Pressing `"` brings up a dialog asking for a target register, that includes the [0-9] capture groups.  So I assumed capture groups were copied to the target registers [0-9]

Would that be a sensible way to handle it? -- persistently store capture groups as registers.  Possibly only overwriting all of the registers when a new capture group is created.  That way I could even do multiple searches, and only loose capture groups when I do a search with a new capture group `()`

**Comment by mawww** [comment]
Registers `0-9` are pretty special yeah, they work completely differently from the other registers. I am not sure its actually warranted, I think we would get all the desired behaviours using regular "static" register semantics for them, while improving Kakoune performances as selections wont have to carry their captures around.

Except for multiple windows... The one thing we would loose is that as the captures wont be stored in the selections anymore, but in a global register, searching in one window will override the values for another window. I am not sure if that would be acceptable or not.

**Comment by samuel1** [comment]
I guess the semantics are similar to the `.` and `#` registers, and those both make sense

On the flip side, the `"` yank register semantics are persistent and work across multiple windows. Both of those are useful properties -- being able to yank something and later paste it in another window

In general I like the idea of being able to yank capture groups too, and later manipulate them, or paste them into other windows

But I can see the dichotomy... Maybe `[0-9]` for yanked capture groups, and `a-[0-9]` for selection capture groups... Or only yanking creates captures??

An alternative might be a global yank/capture-group/selection register history (does it already exist?). That way any window could access all registers and past selections from all other windows

**Comment by TeddyDD** [comment]
Just got bitten by this behaviour. With buffer
```
type foo
```

```
/type (foo)<ret>a<c-r>1<ret> # works
/type (foo)<ret>o<c-r>1<ret> # does not
```
